oneoffs/heatmap.py

```diff
@@ -0,0 +1,86 @@
+"""
+Used to plot a heatmap of the policy and value networks.
+Check FLAGS for default values for what models to load.
+
+Usage:
+python heatmap.py
+
+"""
+import sys; sys.path.insert(0, '.')
+
+import itertools
+import os
+
+import numpy as np
+import tensorflow as tf
+from tqdm import tqdm
+
+import go
+import oneoff_utils
+
+
+tf.app.flags.DEFINE_string(
+    "sgf_dir", "sgf/baduk_db/", "sgf database")
+
+tf.app.flags.DEFINE_string("model_dir", "saved_models",
+                           "Where the model files are saved")
+tf.app.flags.DEFINE_string("data_dir", "data/eval", "Where to save data")
+tf.app.flags.DEFINE_integer("idx_start", 150,
+                            "Only take models after given idx")
+tf.app.flags.DEFINE_integer("eval_every", 5,
+                            "Eval every k models to generate the curve")
+
+FLAGS = tf.app.flags.FLAGS
+
+
+def eval_for_policy(eval_positions, model_dir, data_dir, idx_start, eval_every):
+  model_paths = oneoff_utils.get_model_paths(model_dir)
+
+  print("Evaluating models {}-{}, eval_every={}".format(
+        idx_start, len(model_paths), eval_every))
+  for idx in tqdm(range(idx_start, len(model_paths), eval_every)):
+    if idx == idx_start:
+      player = oneoff_utils.load_player(model_paths[idx])
+    else:
+      oneoff_utils.restore_params(model_paths[idx], player)
+
+    pos_names, positions = zip(*eval_positions)
+    eval_probs, eval_values = player.network.run_many(positions)
```

Comment:
if the positions becomes too large, this call will probably run out of memory.

Also, this will apply a symmetry/rotation transformation at random -- not sure if that's intended or not, but it will make the numbers you get non-repeatable.

Reply:
Yes I'm aware of the rotation problem. Maybe in a later version I'll come up with something better.
I haven't written the batching code because it's not broken for me yet. I can fix it if you deem it important but I'd prefer to punt till later.

Reply:
You can pass in 'false' and it won't apply the transforms.

You don't have to fix it but you should make a note of it in the docstring.